torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1477,6 +1477,19 @@ void LoopNest::reorderAxis(For* a, For* b) {
   For* last = internal_axes.front();
   Stmt* newInner = body;
 
+  s = inner;
+  while (s != outer) {
+    if (auto cond = dynamic_cast<Cond*>(s->get_parent())) {
+      if (s == cond->true_stmt()) {
+        newInner = cond->cloneWithNewBody(newInner);
+      } else {
+        // s is the false branch of Cond
+        newInner = cond->cloneWithNewBodies(new Block({}), newInner);
+      }
+    }
```

Comment:
This should be done iteratively, otherwise we'll hit the same issue if the loop body is
```
if (cond1) {
  if (cond2) {
    a[i] = 0;
  }
}
```
Basically, we should just go up until we hit a `For` and take its body.

Reply:
It is already being done iteratively from `inner` until we reach `outer`. Added a test to make sure the multiple conditions case work as well.